folly/memory/detail/MallocImpl.h

```diff
@@ -38,17 +38,29 @@ int mallctlnametomib(const char*, size_t*, size_t*)
 int mallctlbymib(const size_t*, size_t, void*, size_t*, void*, size_t)
     __attribute__((__nothrow__, __weak__));
 #else
-extern void* (*mallocx)(size_t, int);
-extern void* (*rallocx)(void*, size_t, int);
-extern size_t (*xallocx)(void*, size_t, size_t, int);
-extern size_t (*sallocx)(const void*, int);
-extern void (*dallocx)(void*, int);
-extern void (*sdallocx)(void*, size_t, int);
-extern size_t (*nallocx)(size_t, int);
-extern int (*mallctl)(const char*, void*, size_t*, void*, size_t);
-extern int (*mallctlnametomib)(const char*, size_t*, size_t*);
+// Here the make external declarations consistent with those from jemalloc
+// to avoid issues with different symbol kinds
+#define je_mallocx mallocx
```

Comment:
What happens if these symbols are already defined? If these are defined differently in another included header, then this would be a warning/error (`macro-redefined` for clang, generic for gcc). I don't think we want to be unconditionally `#define`ing names in other namespaces if we can help it.

Also, if they are not already defined, how is this different from before? After preprocessing, the declarations below will not have changed.

Reply:
I read again `memory/Malloc.h` and now I'm back to my first solution.

If JEMALLOC_VERSION is defined in `MallocImpl.h`, then `jemalloc.h` has been included and all these functions have been declared, so no need to do it again.

If jemalloc is used, `jemalloc.h` is included before `MallocImpl.h`, and hence JEMALLOC_VERSION is defined in `MallocImpl.h`.